### DISCUSSION TITLE
Error adding SQL metric: Implicit keys need to be on a single line at line 1, column 1: SELECT ^

### DISCUSSION THREAD

**Discussion by ondar** [cybertec-postgresql/pgwatch#530] [Q&A] [answered]
![image](https://github.com/user-attachments/assets/561dd1e0-ae3e-44e9-869a-a891e9ffbd40)

What am I doing wrong again? 😁

**Comment by pashagolub** [comment, marked as answer]
sqls are maps:
```
"16": >
	select bklablbl
	blajgjg bdgdgjdg
"14": >
	select fosdgnkg 
	egekgengg 
```

![image](https://github.com/user-attachments/assets/5c550988-2b0b-495a-ab6a-b39b0066294f)

**Reply by pashagolub** [reply]
We didn't have time to come up with a dedicated editor. webui is rather ascetic

**Reply by pashagolub** [reply]
@denys-holub FYI

**Reply by ondar** [reply]
Ok, I added at start:

"11": >-

Now I'm getting 500 error and in body:

ERROR: malformed array literal: "["*"]" (SQLSTATE 22P02)

**Reply by ondar** [reply]
That SQL was working, when I was adding it in April through direct edit of file `src/metrics/metrics.yaml`

**Reply by ondar** [reply]
Interesting, seems to be bug again, I got error "ERROR: malformed array literal: "["is_failing_int","seconds_since_last_failure"]" (SQLSTATE 22P02)" saving existing metrics "archiver". So it seems it have something to do with saving the field "Gauges" in tab "SETTINGS".

**Reply by ondar** [reply]
Should I create bug for that?

**Reply by pashagolub** [reply]
Yes, please! File every bug or issue you will find!. Thanks!
